substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -209,6 +200,28 @@ static void println(String str) {
         System.out.println(str);
         // Checkstyle: resume
     }
+
+    private class RuntimeSerializationSupportImpl extends SerializationSupport implements RuntimeSerializationSupport {
```

Comment:
`RuntimeSerializationSupportImpl` must not be an inner class of hosted class `SerializationFeature`. It extends `com.oracle.svm.reflect.serialize.SerializationSupport` which holds data that will be part of an image. We do not allow any hosted classes to get built into an image. If you build an image that uses serialization with image builder assertions enabled (`native-image -J-ea`) you would run into issues with `com.oracle.svm.hosted.NativeImageGenerator#checkName`.

Reply:
This is no longer an inner class.
I was however unable to recreate any issues with `-J-ea` with the previous implementation, so couldn't check if the current implementation mitigates that issue.